File: 2025/Company/Company.Web/Modules/Default/Tenants/TenantsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Company.Default;

[ConnectionKey("Default"), Module("Default"), TableName("Tenants")]
[DisplayName("Tenants"), InstanceName("Tenants")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
public sealed class TenantsRow : Row<TenantsRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Tenant Id"), Identity, IdProperty]
    public int? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    [DisplayName("Tenant Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string TenantName { get => fields.TenantName[this]; set => fields.TenantName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field TenantId;
        public StringField TenantName;

    }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Administration/TenantsRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface TenantsRow {
    TenantId?: number;
    TenantName?: string;
}

export abstract class TenantsRow {
    static readonly idProperty = 'TenantId';
    static readonly nameProperty = 'TenantName';
    static readonly localTextPrefix = 'Administration.Tenants';
    static readonly lookupKey = 'Administration.Tenant';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TenantsRow>('Administration.Tenant') }
    static async getLookupAsync() { return getLookupAsync<TenantsRow>('Administration.Tenant') }

    static readonly deletePermission = 'PermissionKeys.Tenants';
    static readonly insertPermission = 'PermissionKeys.Tenants';
    static readonly readPermission = 'PermissionKeys.Tenants';
    static readonly updatePermission = 'PermissionKeys.Tenants';

    static readonly Fields = fieldsProxy<TenantsRow>();
}